distro/exporter.go

```diff
@@ -49,7 +49,8 @@ func newOTLPExporter(c *exporterConfig) (trace.SpanExporter, error) {
 	if c.TLSConfig != nil {
 		tlsCreds := credentials.NewTLS(c.TLSConfig)
 		opts = append(opts, otlptracegrpc.WithTLSCredentials(tlsCreds))
-	} else {
+	} else if realm, ok := os.LookupEnv(splunkRealmKey); !ok || none(realm) {
+		// use insecure gRPC when SPLUNK_REALM is not set
```

Comment:
To be clear, are you planning to remove the changes not related changing `otlpRealmEndpointFormat`?

Reply:
> Also I suggest not relying on loopback, but rather require explicit distro.WithInsecure when distro.WithEndpoint is used and parse the scheme (http/https) if env var is used. This is what OTel does.

I don't agree with this. Given this distro is intended to work out of the box with a local collector (that does not use TLS) and it is intended to be a much simplified version of the default SDK targeted at Splunk customers it should configured the underlying SDK with minimal configuration requirements put on the customer.

I have a WIP to detect the loopback interface from the endpoint and it doesn't seem unreasonable ([nor unique to this project](https://github.com/upspin/upspin/blob/3bd9dc91d24198addf82b305c06f325d6bb4b936/serverutil/addr.go#L12-L36)) to do so.

Reply:
> To be clear, are you planning to remove the changes not related changing otlpRealmEndpointFormat?

No as it would not fix the bug as the created gRPC client would be "insecure". This is the minimal code to fix the issue.

Reply:
> I don't agree with this. Given this distro is intended to work out of the box with a local collector (that does not use TLS)

The defaults should be still local collector (with insecure). Let me describe the use-cases:

1. `distro.Run()`-> insecure local collector
2. `distro.Run(distro.WithEndpoint("abc:123"))` - secure `abc:123`
3. `distro.Run(distro.WithEndpoint("abc:123"), distro.WithInsecure())` -> insecure `abc:123`
4. `distro.Run()` and `OTEL_EXPORTER_OTLP_ENDPOINT=http://abc:123` -> insecure `abc:123` (AFAIK it is already handled by OTel [here](https://github.com/open-telemetry/opentelemetry-go/blob/4bbf8d6bcf7f3d2765e3a2f7bcc77da00c6babfb/exporters/otlp/otlptrace/internal/otlpconfig/envconfig.go#L58-L84))
5. `distro.Run()` and `SPLUNK_REALM=us0` -> secure `ingest.us0.signalfx.com:443`